src/release-specification.ts

```diff
@@ -210,28 +210,45 @@ export async function validateReleaseSpecification(
     (packageName) =>
       project.workspacePackages[packageName].hasChangesSinceLatestRelease,
   );
-  const missingChangedPackageNames = changedPackageNames.filter(
-    (packageName) =>
-      !hasProperty(unvalidatedReleaseSpecification.packages, packageName) ||
-      unvalidatedReleaseSpecification.packages[packageName] === null,
+  const packageNamesToReportMissing = changedPackageNames.filter(
+    (packageName) => {
+      const isInternalDependency = Object.values(
+        project.workspacePackages,
+      ).some((pkg) => {
+        const { dependencies, peerDependencies } = pkg.validatedManifest;
+        return (
+          (dependencies && hasProperty(dependencies, packageName)) ||
+          (peerDependencies && hasProperty(peerDependencies, packageName))
```

Comment:
I think you can drop the `dependencies &&` and `peerDependencies &&` checks now since we know now they will never be `null` or `undefined`:

``` suggestion
          hasProperty(dependencies, packageName) ||
          hasProperty(peerDependencies, packageName)